src/server/bootstrap/handleKillSignals.js

```diff
@@ -1,34 +1,60 @@
 /* eslint-disable no-console */
 
+import * as services from 'server/services/all'
+
+const SHUTDOWN_TIMEOUT = 3000
+
+let shutdown
+const callbacks = [
+  () => services.disconnect(),
+]
+
+function log(message) {
+  console.info(`${new Date().toJSON()}: ${message}`)
+}
+
 /**
  * terminator === the termination handler
  * Terminate server on receipt of the specified signal.
- * @param {string} sig  Signal to terminate on.
+ * @param {string} signal  Signal to terminate on.
  */
-function terminator(sig) {
-  if (typeof sig === 'string') {
-    console.log(`${Date(Date.now())}: Received ${sig}.`)
-    process.exit(1) // eslint-disable-line no-process-exit
+function terminator(signal) {
+  if (typeof signal === 'string') {
+    log(`Received ${signal}.`)
   }
 
-  console.log(`${Date(Date.now())}: Node server stopped.`)
+  // At the first SIGTERM, we try to shutdown the service gracefully
+  if ((signal === 'SIGTERM' || signal === 'SIGINT') && !shutdown) {
+    log('Shutdown server gracefully...')
+    log(`${SHUTDOWN_TIMEOUT}ms before killing it.`)
+    callbacks.forEach(callback => callback())
+    setTimeout(() => terminator(), SHUTDOWN_TIMEOUT)
+    shutdown = true
+    return
+  }
+
+  process.exit(1) // eslint-disable-line no-process-exit
 }
 
-const handleKillSignals = () => {
+function handleKillSignals() {
   //  Process on exit and signals.
   process.on('exit', () => {
-    terminator()
+    log('Node server stopped.')
   });
 
   // Removed 'SIGPIPE' from the list - bugz 852598.
```

Comment:
Looks like the issue was fixed https://github.com/nodejs/node-v0.x-archive/issues/2286.